Tests/SwiftDocCTests/Semantics/SymbolTests.swift

```diff
@@ -554,12 +560,223 @@ class SymbolTests: XCTestCase {
         
         XCTAssertTrue(unresolvedTopicProblems.contains(where: { $0.diagnostic.localizedSummary == "Topic reference 'doc://com.test.external/ExternalPage' couldn't be resolved. No external resolver registered for 'com.test.external'." }))
         if LinkResolutionMigrationConfiguration.shouldUseHierarchyBasedLinkResolver {
-            XCTAssertTrue(unresolvedTopicProblems.contains(where: { $0.diagnostic.localizedSummary == "Topic reference 'UnresolvableSymbolLinkInMyClassOverview<>(_:))' couldn't be resolved. Reference at '/MyKit/MyClass' can't resolve 'UnresolvableSymbolLinkInMyClassOverview<>(_:))'. No similar pages. Available children: init(), myFunction()." }))
-            XCTAssertTrue(unresolvedTopicProblems.contains(where: { $0.diagnostic.localizedSummary == "Topic reference 'UnresolvableClassInMyClassTopicCuration' couldn't be resolved. Reference at '/MyKit/MyClass' can't resolve 'UnresolvableClassInMyClassTopicCuration'. No similar pages. Available children: init(), myFunction()." }))
+            try XCTAssertElements(in: unresolvedTopicProblems, keyedBy: \.diagnostic.localizedSummary, using: [
+                "Topic reference 'UnresolvableSymbolLinkInMyClassOverview<>(_:))' couldn't be resolved. Reference at '/MyKit/MyClass' can't resolve 'UnresolvableSymbolLinkInMyClassOverview<>(_:))'. No similar pages. Available children: init(), myFunction().": { problem in
+                    XCTAssert(problem.possibleSolutions.isEmpty)
+                },
+                "Topic reference 'UnresolvableClassInMyClassTopicCuration' couldn't be resolved. Reference at '/MyKit/MyClass' can't resolve 'UnresolvableClassInMyClassTopicCuration'. No similar pages. Available children: init(), myFunction().": { problem in
```

Comment:
I find this test helper a bit too unfamiliar which to me makes the test code harder to read. I had to look at the implementation to know that `using` was a dictionary of error messages to closures. 

I would probably find this test easier to read if the code that finds each problem was more explicit. 

The simplest way to do this would be using `first(where:)`:

```swift
let problem = try XCTUnwrap(unresolvedTopicProblems.first(where: { $0.diagnostic.localizedSummary == "Topic reference 'init()' couldn't be resolved. Reference is ambiguous after '/MyKit/MyClass':" }))
```

but another style would be to group the problems in a dictionary and do keyed lookup. 

```swift
let problemsMap = Dictionary(grouping: unresolvedTopicProblems, by: \.diagnostic.localizedSummary)
XCTAssert(problemsMap.values.allSatisfy({ $0.count == 1}), "Unexpected duplicate error messages")

let problem = try XCTUnwrap(problemsMap["Topic reference 'init()' couldn't be resolved. Reference is ambiguous after '/MyKit/MyClass':"]).first
```

Reply:
That's fair. I went with your first suggestion...it's a little repetitive, but I guess that's not necessarily a bad thing in tests.